sfdoc/publish/salesforce.py

```diff
@@ -53,34 +69,68 @@ def _get_salesforce_api(self):
             version=settings.SALESFORCE_API_VERSION,
             client_id='sfdoc',
         )
-        return sf
+        cls.api = sf
 
-    def archive(self, ka_id, kav_id):
+    def get_docsets(self):
+        query_str = f"""SELECT Id, {settings.SALESFORCE_DOCSET_ID_FIELD},
+                    Index_Article_Id__c
+                    FROM {settings.SALESFORCE_DOCSET_SOBJECT}"""
+        return self.api.query(query_str)["records"]
```

Comment:
Keep in mind that query only returns the first batch of results. That's quite a few (defaults to 2000 -- https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/headers_queryoptions.htm?search_text=batch), but 
can be surprising. Use query_all to make multiple requests to collect all results.

Reply:
@davisagli although likely, 2000 is not a guarantee depending on the fields returned and if the object has any special rules.

Reply:
Yeah fair point. The docs are pretty clear that it will vary.

Reply:
In this context, 2000 is two orders of magnitude more than I anticipate, but I added a comment to the code in case someone wants to reevaluate when we get closer to that limit. (e.g. if we were using the system for SFDC instead of SFDO).

ba32b07

Reply:
I realize it's not likely to happen. But my concern is that if it does, it would silently miss some of the records which we could easily overlook. I don't see a problem with proactively switching this to query_all. It'll still be a single request when there aren't other batches to retrieve.

Reply:
Okay thanks for clarifying. I thought that query_all was going to require paging.

d7251aaee6a495b4c6d1c36ec1c6ed7c549afe58